octodns/provider/azuredns.py

```diff
@@ -495,6 +624,324 @@ def _data_for_TXT(self, azrecord):
                                                    ar.value))
                            for ar in azrecord.txt_records]}
 
+    def _data_for_dynamic(self, azrecord):
+        default = set()
+        pools = defaultdict(lambda: {'fallback': None, 'values': []})
+        rules = []
+
+        # top level geo profile
+        geo_profile = self._get_tm_profile_by_id(azrecord.target_resource.id)
+        for geo_ep in geo_profile.endpoints:
+            rule = {}
+
+            # resolve list of regions
+            geo_map = list(geo_ep.geo_mapping)
+            if geo_map != ['WORLD']:
+                if 'GEO-ME' in geo_map:
+                    # Azure treats Middle East as a separate group, but
+                    # its part of Asia in octoDNS, so we need to remove GEO-ME
+                    # if GEO-AS is also in the list
+                    # Throw exception otherwise, it should not happen if the
+                    # profile was generated by octoDNS
+                    if 'GEO-AS' not in geo_map:
+                        msg = '_data_for_dynamic: Profile={}: '.format(
+                            geo_profile.name)
+                        msg += 'Middle East (GEO-ME) is not supported by ' + \
+                               'octoDNS. It needs to be either paired ' + \
+                               'with Asia (GEO-AS) or expanded into ' + \
+                               'individual list of countries.'
+                        raise AzureException(msg)
```

Comment:
Fwiw we transparently mapped `OC` (Oceania) into it's list of countries and back in the NS1 provider, 

https://github.com/octodns/octodns/blob/8be24f10a104c1f238bade6fa71554336bef776e/octodns/provider/ns1.py#L358-L362

It was a PITA and took a lot of care to get all the details right, but just wanted to call that out in case it's of interest to go that route here.

Reply:
Yep I saw that, I found it cleaner to use Asia instead of checking for a static list of countries. Plus we won't run into this unless someone has fiddled with octoDNS-generated Traffic Managers. `octodns-dump` will anyway not work due to this provider expecting certain naming conventions.